src/common/conn/server.rs

```diff
@@ -96,45 +90,83 @@ impl From<&proto::app::v1::CloudConfig> for RobotCloudConfig {
     }
 }
 
-pub struct ViamServerBuilder<'a, M, CC, L, D, C, T> {
+pub struct ViamServerBuilder<'a, T, M, C, CC = WebRtcNoOp, D = WebRtcNoOp, L = NoHttp2> {
     mdns: M,
-    webrtc: Box<WebRtcConfiguration<'a, C, D, CC>>,
+    webrtc: Option<Box<WebRtcConfiguration<'a, D, CC>>>,
     port: u16, // gRPC/HTTP2 port
     http2_listener: L,
     _marker: PhantomData<T>,
     exec: Executor<'a>,
+    app_connector: C,
+    app_config: AppClientConfig,
+}
+
+impl<'a, T, M, C> ViamServerBuilder<'a, T, M, C>
+where
+    M: Mdns,
+    T: AsyncRead + AsyncWrite + Unpin + 'static,
+    C: TlsClientConnector,
+{
+    pub fn new(mdns: M, exec: Executor<'a>, app_connector: C, app_config: AppClientConfig) -> Self {
+        Self {
+            mdns,
+            http2_listener: NoHttp2 {},
+            port: 7888,
+            webrtc: None,
+            _marker: PhantomData,
+            exec,
+            app_connector,
+            app_config,
+        }
+    }
 }
 
-impl<'a, M, CC, L, T, D, C> ViamServerBuilder<'a, M, CC, L, D, C, T>
+impl<'a, T, M, C, CC, D, L> ViamServerBuilder<'a, T, M, C, CC, D, L>
 where
     M: Mdns,
     L: AsyncableTcpListener<T>,
     L::Output: Http2Connector<Stream = T>,
     C: TlsClientConnector,
     D: DtlsBuilder,
-    CC: Certificate,
+    D::Output: 'a,
+    CC: Certificate + 'a,
     T: AsyncRead + AsyncWrite + Unpin + 'static,
 {
-    pub fn new(
-        mdns: M,
-        http2_listener: L,
-        webrtc: Box<WebRtcConfiguration<'a, C, D, CC>>,
-        exec: Executor<'a>,
+    pub fn with_http2<L2, T2>(
```

Comment:
Yes it will be a valid builder but it's type will not be NoHttp2 anymore, of course you *could* invoke the function again passing NoHttp2 hence reversing the action. 

In the builder pattern you have to consume the type to get to the target object, rust would warn about leaving a varaible unassigned (so just calling `my_viam_server_builder.with_http2(listener, 42)`) or reusing it at a later stage will not compile since it was moved at the call